include/envoy/server/hot_restart.h

```diff
@@ -11,6 +13,16 @@ namespace Server {
 
 class Instance;
 
+class OsSysCalls {
+public:
+  virtual ~OsSysCalls(){};
+
+  virtual int shm_open(const char* name, int oflag, mode_t mode) PURE;
```

Comment:
Nit: prefer to Envoy stylify the wrapper names, e.g. `shmOpen`.

Reply:
This I'm torn on.  I like consistent naming (ie making this shmOpen), but its also useful that this is identical to the syscall it mirrors.  Someone looking at the calling code knows exactly what manpage to lookup for semantics by looking at it.  We get no guidance from Envoy::Filesystem::OsSysCalls because that happens to only use 1-word-named syscalls.  

A middle ground could be shmOpen(), with a doc-comment pointing at the real function name and manpage.

Reply:
No real preference from me either way.

Reply:
If we envoy-stylify, is it fTruncate of ftruncate?

Reply:
No preference.